pkg/controller/controller.go

```diff
@@ -348,7 +347,6 @@ func (c *Controller) bootWithError(ctx context.Context) error {
 					return
 				}
 
-				errorGauge.Inc()
```

Comment:
Without context information, this count doesn't add much. While here, I was wondering if we should gracefully exit at this point? In case there's third party error that's not related to reconciled CR, I'd try restarting the operator and if that fails (i.e. operator gets into restart loop) we'd get paged for that. In case of an error here, I believe there's no feasible way to continue anyway?

Reply:
If we don't increment any metric from the error handler, we might as well remove it or is it doing anything that I'm missing?

Reply:
Well, it's logging, but my thinking here was that it might actually make sense to `os.Exit(1)` after logging.